rules/Php80/Rector/Class_/DoctrineTargetEntityStringToClassConstantRector.php

```diff
@@ -0,0 +1,167 @@
+use Rector\Doctrine\NodeAnalyzer\AttributeFinder;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class DoctrineTargetEntityStringToClassConstantRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ClassAnnotationMatcher $classAnnotationMatcher,
+        private readonly AttributeFinder $attributeFinder
+    ) {
+    }
+
+    public function changeTypeInAnnotationTypes(Node $node, PhpDocInfo $phpDocInfo): void
+    {
+        $doctrineAnnotationTagValueNode = $phpDocInfo->getByAnnotationClasses([
+            'Doctrine\ORM\Mapping\OneToMany',
+            'Doctrine\ORM\Mapping\ManyToOne',
+            'Doctrine\ORM\Mapping\OneToOne',
+            'Doctrine\ORM\Mapping\ManyToMany',
+            'Doctrine\ORM\Mapping\Embedded',
+        ]);
+
+        if (! $doctrineAnnotationTagValueNode instanceof DoctrineAnnotationTagValueNode) {
+            return;
+        }
+
+        $this->processDoctrineToMany($doctrineAnnotationTagValueNode, $node);
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Convert targetEntities defined as String to <class>::class Constants in Doctrine Entities.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity="AnotherClass")
+     */
+    private readonly ?Collection $items;
+
+    #[ORM\ManyToOne(targetEntity: "AnotherClass")]
+    private readonly ?Collection $items2;
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity=\Rector\Tests\Php80\Rector\Class_\DoctrineTargetEntityStringToClassConstantRector\Source\AnotherClass::class)
+     */
+    private readonly ?Collection $items;
+
+    #[ORM\ManyToOne(targetEntity: \Rector\Tests\Php80\Rector\Class_\DoctrineTargetEntityStringToClassConstantRector\Source\AnotherClass::class)]
+    private readonly ?Collection $items2;
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        // The minimum Version is PHP 5.5, because we need classname constants,
+        // and support Annotations as well as Attributes.
+//        return PhpVersionFeature::ATTRIBUTES;
+        return PhpVersionFeature::CLASSNAME_CONSTANT;
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+        if ($phpDocInfo) {
+            $this->changeTypeInAnnotationTypes($node, $phpDocInfo);
+        }
+
+        return $this->changeTypeInAttributeTypes($node);
```

Comment:
`$phpDocInfo` has changed check need to be by passed to `return $this->changeTypeInAttributeTypes($node);` so in `changeTypeInAttributeTypes()`, it return `$node` if `$phpDocInfo->hasChanged()`:

```suggestion
        return $this->changeTypeInAttributeTypes($node, $phpDocInfo->hasChanged());
```